gulpfile.js

```diff
@@ -1,210 +1,182 @@
-var path = require("path");
-var webpack_stream = require('webpack-stream');
-var ON_DEATH = require('death')({ debug: false });
-
-var watch = require("gulp-watch");
-var del = require('del');
-var sass = require('gulp-sass');
-var openfinLauncher = require('openfin-launcher');
-var configPath = path.join(__dirname, '/configs/finConfig.json');
-//new
-var StartupConfig = require("./configs/other/server-environment-startup");
-const webpack = require('webpack');
-var chalk = require('chalk');
+// #region Imports
+// NPM
+const chalk = require("chalk");
+const { exec, spawn } = require("child_process");
+const ON_DEATH = require("death")({ debug: false });
+const del = require("del");
+const gulp = require("gulp-4.0.build");
+const sass = require("gulp-sass");
+const watch = require("gulp-watch");
+const merge = require("merge-stream");
+const launcher = require("openfin-launcher");
+const path = require("path");
+const webpack = require("webpack-stream");
+
+// local
+const webpackFilesConfig = require("./build/webpack/webpack.files.js")
+const webpackServicesConfig = require("./build/webpack/webpack.services.js")
+// #endregion
+
+// #region Constants
+const componentsToBuild = require("./build/webpack/webpack.files.entries");
+const startupConfig = require("./configs/other/server-environment-startup");
+
 chalk.enabled = true;
-var serverOutColor = chalk.yellow;
-var errorOutColor = chalk.red;
-var webpackOutColor = chalk.cyan;
-var initialBuildFinished = false;
-
-var componentsToBuild = require('./build/webpack/webpack.files.entries.json');
-
-function buildComponentIgnore() {//Dont copy files that we build
-	var componentIgnores = [];
-	var components = componentsToBuild;
-	for (var key in components) {
-		var filename = components[key].entry.split("/").pop();
-		componentIgnores.push(path.join('!' + __dirname, components[key].entry));
-	}
-	return componentIgnores;
+const errorOutColor = chalk.red;
+// #endregion
+
+// #region Script variables
+let distPath = path.join(__dirname, "dist");
+let srcPath = path.join(__dirname, "src");
+
+let initialBuildFinished = false;
+// #endregion
+
+// #region Functions
+/**
+ * Cleans the project folder of generated files.
+ */
+const clean = () => {
+	return del(distPath, { force: true });
 }
 
-function copyStaticComponentsFiles() {
-	var source = [path.join(__dirname, '/src/components/**/*'), path.join('!' + __dirname, '/src/components/**/*.jsx')]
-	source = source.concat(buildComponentIgnore());
-	return gulp.src(source)
-		.pipe(gulp.dest(path.join(__dirname, '/dist/components/')));
+/** 
+ * Copies static files to the output directory.
+ */
+const copyStaticFiles = () => {
+	const source = [
+		path.join(srcPath, "components", "**", "*"),
+		"!" + path.join(srcPath, "components", "**", "*.jsx")];
+
+	// Don't copy files that we build
+	for (const key in componentsToBuild) {
+		source.push("!" + path.join(__dirname, componentsToBuild[key].entry));
+	}
+	
+	return merge(
```

Comment:
what's merge? does it merge the streams into one?

Reply:
Yes, so the task doesn't complete until all of the streams complete. https://www.npmjs.com/package/merge-stream

Reply:
looks like it. disregard